app/src/main/cpp/BrowserWorld.cpp

```diff
@@ -1911,11 +1909,33 @@ BrowserWorld::CreateSkyBox(const std::string& aBasePath, const std::string& aExt
 
   if (extension == ".ktx") {
 #if defined(OPENXR) && defined(OCULUSVR)
-    glFormat =  GL_COMPRESSED_SRGB8_ETC2;
+    glFormat = GL_COMPRESSED_SRGB8_ETC2;
 #else
-    glFormat =  GL_COMPRESSED_RGB8_ETC2;
+    glFormat = GL_COMPRESSED_RGB8_ETC2;
 #endif
   }
+
+#if defined(OPENXR)
+  // Check if the device supports the requested format. If not, use the other one.
+  std::vector<int64_t> availableGLFormats = m.device->GetAvailableColorFormat();
+  if (!availableGLFormats.empty() && std::find(availableGLFormats.begin(), availableGLFormats.end(), glFormat)
+    == availableGLFormats.end()) {
+    if (extension == ".ktx") {
```

Comment:
I think we can simplify this code with some kind of data structure to define device specific capabilities; something like this perhaps ?

device-type-id: {
  extension: 
    {  target,  default  }
}  

I think we also have pre-defined extensions for each device, so perhaps we could simplify this data structure even more.

Reply:
Let's use the control flow instead of this data structure. As there will be a lot of the same `{ target, default }` for different devices, control flow like this will make things clear: https://github.com/Igalia/wolvic/pull/859/files#diff-95306c461487009203455d0e541c700198235f0d649f9acec07a4fc720a59a8aR291-R326

Feel free to add specific logic since I don't have devices other than Meta Quest 2.